src/Engine/Renderer/Material/MaterialConverters.hpp

```diff
@@ -0,0 +1,83 @@
+#ifndef RADIUMENGINE_MATERIALCONVERTERS_HPP
+#define RADIUMENGINE_MATERIALCONVERTERS_HPP
+
+#include <Engine/RaEngine.hpp>
+
+#include <Core/File/MaterialData.hpp>
+
+#include <functional>
+#include <string>
+
+namespace Ra {
+  namespace Engine {
+    class Material;
+  }
+}
+
+
+/// A material converter is a couple <std::string, std::function<Ra::Engine::Material*(Ra::Asset::MaterialData*)>
+/// The string gives the mname of the material, the function is watever is compatible wit std::function :
+///     - a lambda
+///     - a functor
+///     - a function with bind parameteres ....
+/// The function is in charge of converting a concrete Ra::Asset::MaterialData* to a concrete Ra::Engine::Material*
+/// according to the type of material described by the string ...
+/**
+ * Instruction on how to extend the material system
+ */
+///////////////////////////////////////////////
+////        Material converter system       ///
+///////////////////////////////////////////////
+
+namespace Ra {
+  namespace Engine {
+    namespace MaterialConverterSystem {
+
```

Comment:
Why "System"?

Reply:
Change to EngineMaterialConverters (to be consistant with EngineRenderTechniques)